lib/libimhex/include/hex/pattern_language/ast_node.hpp

```
@@ -1408,6 +1409,15 @@ namespace hex::pl {
 
             structCleanup.release();
 
+            if (!pattern->hasOverriddenColor()) {
+                if (auto value = getAttributeValue("color"); value) {
+                    u32 color = strtoul(value->c_str(), nullptr, 16);
+                    pattern->setColor(hex::changeEndianess(color, std::endian::big) >> 8);
+                } else if (auto value = hasAttribute("single_color", false); value) {
```

Should this be just an `if` instead of an `else if`? If you set both the `color` and `single_color` attribute for a struct, `single_color` will be silently ignored

---

`color` and `single_color` do the same thing in the end, and I wanted for `color` to be always a priority, since user explicitly sets color instead or relying on the next color from color palette like `single_color` does.